openedx/core/djangoapps/schedules/signals.py

```diff
@@ -61,6 +61,24 @@ def create_schedule(sender, **kwargs):  # pylint: disable=unused-argument
         ))
 
 
+@receiver(ENROLL_STATUS_CHANGE)
+def update_schedule(sender, event, user, course_id, **kwargs):  # pylint: disable=unused-argument
+    """
+    When a CourseEnrollment's status is updated, update the Schedule's active status if configured.
+    """
+    try:
+        schedule = Schedule.objects.get(enrollment__user=user, enrollment__course=course_id)
+    except Schedule.DoesNotExist:
+        # Exit since it could just be an indication of Schedules are not enabled.
+        return
+
+    if event == 'enroll':
+        schedule.active = True
+    elif event == 'unenroll':
+        schedule.active = False
```

Comment:
Such direct coupling makes me wonder why we have an `active` field on Schedule at all? Rather than just using `enrollment__is_active`...

Reply:
The original use was to be able to deactivate customized schedules during Dynamic Pacing rollout, I believe. We likely don't need that column anymore, since schedules are a more tightly integrated feature.

Reply:
Created https://openedx.atlassian.net/browse/AA-488 in hopes it'll happen one day, but want to get this fix out there without waiting through the multiple PRs for removing a field (and a migration on Friday afternoon to boot)